modules/pricer/src/main/java/com/opengamma/strata/pricer/credit/cds/IsdaCdsTradePricer.java

```diff
@@ -0,0 +1,325 @@
+
+import java.time.LocalDate;
+
+import com.opengamma.strata.basics.ReferenceData;
+import com.opengamma.strata.basics.currency.CurrencyAmount;
+import com.opengamma.strata.basics.currency.Payment;
+import com.opengamma.strata.market.sensitivity.PointSensitivityBuilder;
+import com.opengamma.strata.pricer.DiscountingPaymentPricer;
+import com.opengamma.strata.pricer.common.PriceType;
+import com.opengamma.strata.product.credit.ResolvedCds;
+import com.opengamma.strata.product.credit.ResolvedCdsTrade;
+
+/**
+ * Pricer for single-name credit default swaps (CDS) trade based on ISDA standard model. 
+ * <p>
+ * The implementation is based on the ISDA model versions 1.8.2.
+ */
+public class IsdaCdsTradePricer {
+
+  /**
+   * Default implementation.
+   */
+  public static final IsdaCdsTradePricer DEFAULT = new IsdaCdsTradePricer();
+
+  /**
+   * The product pricer.
+   */
+  private final IsdaCdsProductPricer productPricer;
+  /**
+   * The upfront fee pricer.
+   */
+  private final DiscountingPaymentPricer upfrontPricer;
+
+  //-------------------------------------------------------------------------
+  /**
+   * The default constructor.
+   * <p>
+   * The default pricers are used.
+   */
+  public IsdaCdsTradePricer() {
+    this.productPricer = IsdaCdsProductPricer.DEFAULT;
+    this.upfrontPricer = DiscountingPaymentPricer.DEFAULT;
+  }
+
+  /**
+   * The constructor with the accrual-on-default formula specified.
+   * 
+   * @param formula  the accrual-on-default formula
+   */
+  public IsdaCdsTradePricer(AccrualOnDefaultFormula formula) {
+    this.productPricer = new IsdaCdsProductPricer(formula);
+    this.upfrontPricer = DiscountingPaymentPricer.DEFAULT;
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Calculates the price of the underlying product, which is the present value per unit notional. 
+   * <p>
+   * This is coherent to {@link #presentValueOnSettle(ResolvedCds, CreditRatesProvider, PriceType, ReferenceData)}.
+   * 
+   * @param trade  the trade
+   * @param ratesProvider  the rates provider
+   * @param referenceDate  the reference date
+   * @param priceType  the price type
+   * @param refData  the reference data
+   * @return the price
+   */
+  public double price(
+      ResolvedCdsTrade trade,
+      CreditRatesProvider ratesProvider,
+      PriceType priceType,
+      ReferenceData refData) {
+
+    return price(trade, ratesProvider, trade.getProduct().getFixedRate(), priceType, refData);
+  }
+
+  // internal price computation with specified coupon rate
+  double price(
+      ResolvedCdsTrade trade,
+      CreditRatesProvider ratesProvider,
+      double flactionalSpread,
+      PriceType priceType,
+      ReferenceData refData) {
+
+    ResolvedCds product = trade.getProduct();
+    LocalDate settlementDate = trade.getInfo().getSettlementDate()
+        .orElse(product.getSettlementDateOffset().adjust(ratesProvider.getValuationDate(), refData));
```

Comment:
Add `ResolvedCds.calculateSettlementDateFromValuationDate(valDate)`, storing the calendar in `ResolvedCds`